### ISSUE TITLE
Inconsistent conventions in function names

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#2781]
### Describe the bug

Here the points go second
https://kornia.readthedocs.io/en/latest/geometry.linalg.html#kornia.geometry.linalg.transform_points

Here first
https://kornia.readthedocs.io/en/latest/geometry.camera.perspective.html

### Reproduction steps

```bash
https://twitter.com/Parskatt/status/1754501114777456916
```

### Expected behavior

Not be confusing.

### Environment

```shell
N/a
```

**Comment by edgarriba** [comment]
what is the confusion here ? even though you can use one to solve the other, in principle the functions in principle are unrelated.

The first, applies a generic transformation to coordinates system of a set points given the mapping transformation
```python
points_in_robot = transform_points(robot_from_world, points_in_world) 
```

the second has a specific purpose to project a set of 3d points to the z1 image plane given the calibration parameters of an affine model. The last is an old function that's is planned to be replaced by the new cameras api (based on sophus conventions): https://kornia.readthedocs.io/en/latest/geometry.camera.html

**Comment by ducha-aiki** [comment]
Well, it would be really simple and great, if we use same idea - points first, or points last, across all the library. 
Otherwise, one has to consistently look at the documentation. 
Old or not, it doesn't matter.

**Comment by Parskatt** [comment]
https://twitter.com/Parskatt/status/1757423036771254499

**Comment by Parskatt** [comment]
This inconsistency makes it confusing to use, @edgarriba  

To me, projection is P: 3x4, intrinsics are K: 3x3, and pose is R: 3x3 ,t: 3x1 alt. T: 4x4

```unproject_points``` should take K + image_coords + depth -> 3D points
```transform_points``` should take 3D points + either (R, t) or T -> 3D points
```projection_from_KRt``` takes K + R + t -> P
```project_points``` should take 3D points + P -> 2D points

For me this is by far the most common usecase of these functions. Am I supposed to be creating ```PinholeCamera``` objects? To me the functional form is much easier to read.

**Comment by ducha-aiki** [comment]
@edgarriba exactly, we have inconsistent inspirations, and we should unify them. Users don't care about the history

**Comment by Parskatt** [comment]
While I'm at it, why is this:
![Screenshot_20240214_115137_Chrome.jpg](https://github.com/kornia/kornia/assets/22053118/49e1f778-ab12-42bb-baa6-3212970d22c3)

Composed on the left? I expected it would be on the right.

**Comment by edgarriba** [comment]
> Composed on the left? I expected it would be on the right.

that was an arbitrary choice i guess when that function was created. Depending on the application you would like in one or the other side, I'd be in favor in deprecating that and just teaching our users to use as follows:

these all should be equivalent (rigid body transform and points conventions are explained [here](https://farm-ng.github.io/farm-ng-core/docs/Developer%20Guide/poses_and_frames))

```python
t1_from_t2 = compose_transform(t1_from_t0, t0_from_t2)
t1_from_t2 = compose_transform(t0_from_t1.inverse(), t0_from_t2)
t1_from_t2 = compose_transform(t1_from_t0, t2_from_t0.inverse())
t1_from_t2 = compose_transform(t0_from_t1.inverse(), t2_from_t0.inverse())
t1_from_t2 = compose_transform(t2_from_t0, t1_from_t0.inverse()).inverse()
```

where `compose_transform` is just a convenient/tested function that handles `matmul` and batching if existing. That's my plan at least for kornia-rs which eventually will be the backend for small rank matrices. Pytorch is extremely inefficient at that [link1](https://discuss.pytorch.org/t/matmul-slow-for-small-matrices/168425) [link2](https://discuss.pytorch.org/t/pytorch-tensor-slow-on-small-matrix-operations-large-overhead/67882/6)

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
